miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/QcRestControllerST.java

```diff
@@ -0,0 +1,135 @@
+import uk.ac.bbsrc.tgac.miso.core.data.qc.SampleQC;
+import uk.ac.bbsrc.tgac.miso.core.data.qc.QC;
+import uk.ac.bbsrc.tgac.miso.core.data.qc.QcTarget;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.QcDto;
+import static org.hamcrest.Matchers.*;
+
+import static org.junit.Assert.*;
+import org.springframework.test.web.servlet.MockMvc;
+
+
+public class QcRestControllerST extends AbstractST {
+
+  private static final String CONTROLLER_BASE = "/rest/qcs";
+  private static final Class<SampleQC> entityClass = SampleQC.class;
+  // there's library, pool, and sample QC entities
+  // they're mostly the same, save for their list of control runs and a field referencing the entity
+  // that they are attached to
+
+  // given this, it's acceptable to just test SampleQCs
+
+
+  private List<QcDto> makeCreateDtos() {
+    QcDto dto1 = new QcDto();
+    dto1.setEntityId(1L);
+    dto1.setDate("2025-07-30");
+    dto1.setQcTypeId(101L);
+    dto1.setResults("5.4");
+    dto1.setQcTarget("Sample");
+
+
+    QcDto dto2 = new QcDto();
+    dto2.setEntityId(2L);
+    dto2.setDate("2025-05-05");
+    dto2.setQcTypeId(102L);
+    dto2.setResults("9.8");
+    dto2.setQcTarget("Sample");
+
+    return Arrays.asList(dto1, dto2);
+  }
+
+  @Test
+  public void testBulkCreateAsync() throws Exception {
+    List<SampleQC> qcs = baseTestBulkCreateAsync(CONTROLLER_BASE, entityClass, makeCreateDtos());
+    assertEquals(1L, qcs.get(0).getEntity().getId());
+    assertEquals("2025-07-30", qcs.get(0).getDate().toString());
+    assertEquals(101L, qcs.get(0).getType().getId());
+    assertEquals(5.4, qcs.get(0).getResults().doubleValue(), 0.0);
+    assertEquals(QcTarget.Sample, qcs.get(0).getType().getQcTarget());
+
+    assertEquals(2L, qcs.get(1).getEntity().getId());
+    assertEquals("2025-05-05", qcs.get(1).getDate().toString());
+    assertEquals(102L, qcs.get(1).getType().getId());
+    assertEquals(9.8, qcs.get(1).getResults().doubleValue(), 0.0);
+    assertEquals(QcTarget.Sample, qcs.get(1).getType().getQcTarget());
+
+  }
+
+  @Test
+  public void testBulkUpdateAsync() throws Exception {
+    QcDto dto1 = Dtos.asDto(currentSession().get(entityClass, 1));
+    QcDto dto2 = Dtos.asDto(currentSession().get(entityClass, 2));
+
+    dto1.setDescription("updated one");
+    dto1.setQcTarget("Sample");
+    dto2.setDescription("updated two");
+    dto2.setQcTarget("Sample");
+
+    // setting QC target is required for the request to go through properly
+
+    List<SampleQC> qcs =
+        (List<SampleQC>) baseTestBulkUpdateAsync(CONTROLLER_BASE, entityClass, Arrays.asList(dto1, dto2),
+            Arrays.asList(1, 2));
+    assertEquals(dto1.getDescription(), qcs.get(0).getDescription());
+    assertEquals(dto2.getDescription(), qcs.get(1).getDescription());
+
+  }
+
+  @Test
+  public void testDelete() throws Exception {
+    // only admin or owner can delete
+    assertNotNull(currentSession().get(entityClass, 2));
```

Comment:
I'm inferring that whichever user the test is running as happens to be the owner of SampleQC 2? It would be more obvious/less inference necessary if you used the same data in the `testDelete` and `testDeleteFail` and only changed which user was running the operation (like you have in previous delete tests, where you've run them as admin to succeed)